src/main/java/com/minecolonies/coremod/util/BarbarianUtils.java

```diff
@@ -0,0 +1,303 @@
+        citizensList.addAll(colony.getCitizens().values());
+
+        for (@NotNull final CitizenData citizen : citizensList)
+        {
+            if (citizen.getJob() != null && citizen.getWorkBuilding() != null)
+            {
+                final int buildingLevel = citizen.getWorkBuilding().getBuildingLevel();
+                levels += buildingLevel;
+            }
+        }
+
+        if (colony.getTownHall() != null)
+        {
+            return (levels + colony.getTownHall().getBuildingLevel());
+        }
+        else
+        {
+            return 0;
+        }
+    }
+
+    /**
+     * Simple method that returns whether or not an entity is a barbarian.
+     *
+     * @param entity The entity to check
+     * @return Boolean value of whether the entity is a barbarian
+     */
+    public static Boolean isBarbarian(final Entity entity)
+    {
+        return (entity instanceof EntityBarbarian || entity instanceof EntityArcherBarbarian || entity instanceof EntityChiefBarbarian);
+    }
+
+    /**
+     * Returns the closest barbarian to an entity.
+     *
+     * @param entity             The entity to test against
+     * @param distanceFromEntity The distance to check for
+     * @return the barbarian (if any) that is nearest
+     */
+    public static Entity getClosestBarbarianToEntity(final Entity entity, final double distanceFromEntity)
+    {
+        final List<Entity> entityList = CompatibilityUtils.getWorld(entity).getEntitiesInAABBexcluding(
+          entity,
+          entity.getEntityBoundingBox().expand(
+            distanceFromEntity,
+            3.0D,
+            distanceFromEntity),
+          Entity::isEntityAlive);
+
+        final Optional<Entity> entityBarbarian = entityList.stream()
+                                                   .filter(BarbarianUtils::isBarbarian)
+                                                   .findFirst();
+
+        return entityBarbarian.orElse(null);
+    }
+
+    /**
+     * Returns the barbarians close to an entity.
+     *
+     * @param entity             The entity to test against
+     * @param distanceFromEntity The distance to check for
+     * @return the barbarians (if any) that is nearest
+     */
+    public static Stream<EntityLivingBase> getBarbariansCloseToEntity(final Entity entity, final double distanceFromEntity)
+    {
+        final List<EntityLivingBase> entityList = CompatibilityUtils.getWorld(entity).getEntitiesWithinAABB(
+          EntityLivingBase.class,
+          entity.getEntityBoundingBox().expand(
+            distanceFromEntity,
+            3.0D,
+            distanceFromEntity),
+          Entity::isEntityAlive);
+
+        return entityList.stream().filter(BarbarianUtils::isBarbarian);
+    }
+
+    /**
+     * Enacts the Raid Event.
+     *
+     * @param colony The colony for which the raid is occurring in.
+     */
+    public static void eventRaid(final Colony colony)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') The Cyclomatic Complexity of this method "eventRaid" is 20 which is greater than 15 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)